### PR TITLE
Abort draft list if invalid framework is provided

### PR DESCRIPTION
**Pull request by robyoung**

### CODE REVIEW

File: app/main/views/drafts.py [hunk 1]
```diff
@@ -139,6 +139,8 @@ def list_draft_services():
         framework = Framework.query.filter(
             Framework.slug == framework_slug
         ).first()
```
Comment (allait) [inline app/main/views/drafts.py]: `.first_or_404` would've worked just fine, I think we use it quite a lot in other places

**Comment by allait** [comment]
:+1:

File: app/main/views/drafts.py (hunk 1 above)
Comment (robyoung) [inline app/main/views/drafts.py]: The reason I didn't use `first_or_404` here was that the error message didn't give any indication as to what had gone wrong. Given that multiple things could be causing the 404 I think a specific error message is more useful.

Comment (robyoung) [inline app/main/views/drafts.py]: It's a bit of a pain that you can't provide an override error message to `first_or_404`.

**Pull request merged**